crates/libtiny_tui/src/tests/resize.rs

```diff
@@ -0,0 +1,267 @@
+    for _ in 0..2 {
+        tui.handle_input_event(term_input::Event::Key(Key::ShiftUp), &mut None);
+    }
+    tui.draw();
+    tui.set_size(18, 11);
+    tui.draw();
+
+    #[rustfmt::skip]
+    let screen2 =
+       "|osa1: line5       |
+        |osa1: line6       |
+        |osa1: line7       |
+        |osa1: line8       |
+        |osa1: line9       |
+        |osa1: line10      |
+        |osa1: line11      |
+        |osa1: line12      |
+        |osa1: line13      |
+        |                  |
+        |< #chan           |";
+    expect_screen(screen2, &tui.get_front_buffer(), 18, 11, Location::caller());
+}
+
+#[test]
+fn test_resize_no_scroll_stay_on_bottom() {
+    let mut tui = TUI::new_test(18, 10);
+    let serv = "irc.server_1.org";
+    let chan = ChanNameRef::new("#chan");
+    tui.new_server_tab(serv, None);
+    tui.set_nick(serv, "osa1");
+    tui.new_chan_tab(serv, chan);
+    tui.next_tab();
+    tui.next_tab();
+
+    let target = MsgTarget::Chan { serv, chan };
+    let ts = time::at_utc(time::Timespec::new(0, 0));
+
+    for i in 0..15 {
+        tui.add_privmsg("osa1", &format!("line{}", i), ts, &target, false, false);
+    }
+
+    tui.draw();
+
+    #[rustfmt::skip]
+    let screen1 =
+       "|osa1: line7       |
+        |osa1: line8       |
+        |osa1: line9       |
+        |osa1: line10      |
+        |osa1: line11      |
+        |osa1: line12      |
+        |osa1: line13      |
+        |osa1: line14      |
+        |                  |
+        |< #chan           |";
+
+    expect_screen(screen1, &tui.get_front_buffer(), 18, 10, Location::caller());
+
+    tui.set_size(18, 11);
+    tui.draw();
+
+    // shows one extra line on the top of the screen
+    #[rustfmt::skip]
+    let screen2 =
+       "|osa1: line6       |
+        |osa1: line7       |
+        |osa1: line8       |
+        |osa1: line9       |
+        |osa1: line10      |
+        |osa1: line11      |
+        |osa1: line12      |
+        |osa1: line13      |
+        |osa1: line14      |
+        |                  |
+        |< #chan           |";
+    expect_screen(screen2, &tui.get_front_buffer(), 18, 11, Location::caller());
+
+    // resize back to original screen and verify last line is still on the bottom
+    tui.set_size(18, 10);
+    tui.draw();
+    expect_screen(screen1, &tui.get_front_buffer(), 18, 10, Location::caller());
+
+    tui.add_privmsg("osa1", "line15", ts, &target, false, false);
+    tui.set_size(18, 11);
+    tui.draw();
+
+    #[rustfmt::skip]
+    let screen3 =
+       "|osa1: line7       |
+        |osa1: line8       |
+        |osa1: line9       |
+        |osa1: line10      |
+        |osa1: line11      |
+        |osa1: line12      |
+        |osa1: line13      |
+        |osa1: line14      |
+        |osa1: line15      |
+        |                  |
+        |< #chan           |";
+    expect_screen(screen3, &tui.get_front_buffer(), 18, 11, Location::caller());
+}
+
+#[test]
+fn test_resize() {
```

Comment:
What is this testing exactly? Is it checking that it doesn't crash?

Reply:
Yeah it looks like it tests resizing the window to very small (0x0). I think it was added when we were having crashes like that from all my changes 😅

Reply:
In that case why not test the whole range of `[(0,0), (N, N)]`? Have a nested loop for (x, y) pairs starting from (0, 0) to (N, N) and resize in the loop body. Code should be smaller too.